src/structures/paging/mapper/mod.rs

```diff
@@ -398,6 +660,50 @@ impl<S: PageSize> MapperFlush<S> {
     pub fn ignore(self) {}
 }
 
+/// This type represents a range of pages whose mappings have changed in the page table.
+///
+/// The old mappings might be still cached in the translation lookaside buffer (TLB), so they need
+/// to be flushed from the TLB before they're accessed. This type is returned from a function that
+/// changed the mappings of a range of pages to ensure that the TLB flush is not forgotten.
+#[derive(Debug)]
+#[must_use = "Page Table changes must be flushed or ignored."]
+pub struct MapperFlushRange<S: PageSize>(PageRange<S>);
+
+impl<S: PageSize> MapperFlushRange<S> {
+    /// Create a new flush promise
+    #[inline]
+    fn new(pages: PageRange<S>) -> Self {
+        MapperFlushRange(pages)
+    }
+
+    /// Create a new empty flush promise
+    #[inline]
+    fn empty() -> Self {
+        MapperFlushRange::new(PageRange {
+            start: Page::containing_address(VirtAddr::zero()),
+            end: Page::containing_address(VirtAddr::zero()),
+        })
+    }
+
+    /// Flush the page from the TLB to ensure that the newest mapping is used.
+    #[cfg(feature = "instructions")]
+    #[inline]
+    pub fn flush(self) {
```

Comment:
Depending of the number of elements in the range, it might be more efficient to just flush the TLB completely. Maybe this type could provide methods for both approaches, e.g. `flush_ranges` and `flush_all`. Then the users can decide what they want to do, e.g. depending on the range size.